### PR TITLE
feat: add option to specify time range in predictions

### LINKED ISSUE #326: feat: add option to specify time range in predictions
Creating features in bins of e.g. 0-7, 7-30, 30-90, 90-365, 365-... days from prediction time instead of always going from 0-n as we do now, could potentially keep more temporal information and create better predictors.

### PR DESCRIPTION
**Pull request by HLasse**
Fixes #326.

## Notes for reviewers
Adds functionality for specifying time intervals to look behind/ahead in both predictor and outcome specs. 

If specified as float | int, behaviour is as usual. 

The generated col names are made to be backwards compatible. If min_days is 0, the col name will be created as normal, if not, the interval will be added, e.g. pred|outc_value_within_ **0_to_2** _days_maximum_fallback_nan_dichotomous

Tagging for awareness @Gotticketsto360tour

### CODE REVIEW

File: src/timeseriesflattener/feature_specs/single_specs.py [hunk 1]
```diff
@@ -92,18 +119,32 @@ class OutcomeSpec(BaseModel):
...
+    lookahead_days: Union[float, Tuple[float, float]]
     aggregation_fn: AggregationFunType
     fallback: Union[float, int]
     incident: bool
     prefix: str = "outc"
 
+    def __post_init__(self):
+        if isinstance(self.lookahead_days, (float, int)):
+            self.lookahead_days = (0, self.lookahead_days)
+        check_lookperiod_tuple_is_valid(self.lookahead_days)
```
Comment (MartinBernstorff) [inline src/timeseriesflattener/feature_specs/single_specs.py:131]: I'd move this into the lookahead_period property 👍

File: src/timeseriesflattener/feature_specs/single_specs.py [hunk 2]
```diff
@@ -143,15 +182,29 @@ class PredictorSpec(BaseModel):
     feature_base_name: str
     aggregation_fn: AggregationFunType
     fallback: Union[float, int]
-    lookbehind_days: float
+    lookbehind_days: Union[float, Tuple[float, float]]
```
Comment (MartinBernstorff) [inline src/timeseriesflattener/feature_specs/single_specs.py:185]: This type is used in so many places that it perhaps should have a name?

File: src/timeseriesflattener/feature_specs/single_specs.py [hunk 3]
```diff
@@ -143,15 +182,29 @@ class PredictorSpec(BaseModel):
...
     prefix: str = "pred"
 
+    def __post_init__(self):
+        if isinstance(self.lookbehind_days, (float, int)):
+            self.lookbehind_days = (0, self.lookbehind_days)
+        check_lookperiod_tuple_is_valid(self.lookbehind_days)
```
Comment (MartinBernstorff) [inline src/timeseriesflattener/feature_specs/single_specs.py:191]: I'd move this into the lookahead_period property 👍

If the logic is identical, perhaps refactor out?

**Review by MartinBernstorff — APPROVED** [review]
Looks great! 

A few comments on style.

Perhaps we'd like a test asserting that the output column names are as we expect as well?

**Comment by Gotticketsto360tour** [comment]
amazing, thanks for this!

**Pull request merged**